presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -274,7 +274,7 @@ protected Optional<InternalNodeInfo> chooseLeastBusyNode(SplitWeight splitWeight
         return Optional.of(new InternalNodeInfo(chosenNode, false));
     }
 
-    private List<InternalNode> getEligibleNodes(int limit, NodeMap nodeMap, List<RemoteTask> existingTasks)
+    private List<InternalNode> getEligibleNodes(int limit, NodeSet nodeMap, List<RemoteTask> existingTasks)
```

Comment:
So where are we on the debate of NodeMap -> NodeSet ? Do we need to make that change ?